contribs/gmf/src/directives/layertree.js

```diff
@@ -0,0 +1,301 @@
+  if (node.type === 'WMTS') {
+    var newLayer = new ol.layer.Tile();
+    this.layerHelper_.setHelperID(newLayer, layerURL, layerName);
+    var promise = this.layerHelper_.createWMTSLayerFromCapabilitites(layerURL,
+        layerName);
+    promise.then(function(layer) {
+      if (goog.isDef(layer)) {
+        newLayer.setSource(layer.getSource());
+      }
+    });
+    layer = newLayer;
+  } else {
+    layer = this.layerHelper_.createBasicWMSLayer(layerURL, layerName);
+  }
+  this.existingLayers_.push(layer);
+  return layer;
+};
+
+
+/**
+ * Remove layer from map and from this gmf-layertree's existing layers when the
+ * relative layertree of the layer catch an on destroy event.
+ * @param {angular.Scope} scope treeCtrl scope.
+ * @param {ngeo.LayertreeController} treeCtrl ngeo layertree controller, from
+ *     the current node.
+ * @export
+ */
+gmf.LayertreeController.prototype.listeners = function(scope, treeCtrl) {
+  var existingLayers = this.existingLayers_;
+  scope.$on('$destroy', angular.bind(treeCtrl, function() {
+    var i, l;
+    // Remove treeCtrl.layer from  map.
+    treeCtrl.map.removeLayer(treeCtrl.layer);
+    // Remove the treeCtrl.layer from gmf layertree's layers.
+    for (i = 0; i < existingLayers.length; i++) {
+      l = existingLayers[i];
+      if (l === treeCtrl.layer) {
+        existingLayers.splice(i, 1);
+        break;
+      }
+    }
+  }));
+};
+
+
+/**
+ * Return 'outOfResolution' if the current resolution of the map is out of
+ * the min/max resolution in the node.
+ * @param {GmfThemesNode} node Layer tree node.
+ * @return {?string} 'outOfResolution' or null.
+ * @export
+ */
+gmf.LayertreeController.prototype.getResolutionStyle = function(node) {
+  var style;
+  var resolution = this.map.getView().getResolution();
+  var maxExtent = node.maxResolutionHint;
+  var minExtent = node.minResolutionHint;
+  if (goog.isDef(minExtent) && resolution < minExtent ||
+      goog.isDef(maxExtent) && resolution > maxExtent) {
+    style = 'outOfResolution';
+  }
+  return style || null;
+};
+
+
+/**
+ * Toggle activation of a node by adding or removing relative(s) layer(s) on
+ * the map.
+ * @param {ngeo.LayertreeController} treeCtrl ngeo layertree controller, from
+ *     the current node.
+ * @export
+ */
+gmf.LayertreeController.prototype.toggleActive = function(treeCtrl) {
+  var layer = treeCtrl.layer;
+  // Check if the current node state is 'activated'.
+  var add = (this.getSetActive(treeCtrl) === 'on') ? false : true;
+  // Add/remove layers
+  var layers = this.layerHelper_.getFlatLayers(layer);
+  this.layerHelper_.moveInOutLayers(this.map, layers, add);
+};
+
+
+/**
+ * Return a class name that match with the current node activation state.
+ * @param {ngeo.LayertreeController} treeCtrl ngeo layertree controller, from
+ *     the current node.
+ * @return {string} 'on' or 'off' or 'indeterminate'.
+ * @export
+ */
+gmf.LayertreeController.prototype.getSetActive = function(treeCtrl) {
```

Comment:
> `getSetActive` name is not appropriate.

Done

> This method seems a bit too complex to be called on each node on each $digest loop.

True but...

> Seems better to have a value that store the group node state.
This value is watched in the template (more simple watch function).
This value is updated each time a node is changed (actually your function is not called on digest, but manually when you need).

The last point is just hard to do. Because a node doesn't know its parent(s). It just knows its layer and its node. So I must keep somewhere and manage all ol.layer.Group objects created in the gmf-layertree.
(And obviously, all node are not updated when one has its layer added on the map)